### PR TITLE
feat: added contracts sdk

### PR DESCRIPTION
**Pull request by Lukasz2891**

### CODE REVIEW

File: packages/sdk/src/contracts/prices/PricesContractAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { ContractDataProvider } from "../ContractDataProvider";
+import { PricesCommandFactory } from "./PricesCommandFactory";
+import { ContractAdapter } from "../ContractAdapter";
+
+export class PricesContractAdapter implements ContractAdapter {
+  constructor(private commandFactory: PricesCommandFactory) {}
+
+  async getPrices(dataProvider: ContractDataProvider): Promise<number[]> {
```
Comment (hatskier) [inline packages/sdk/src/contracts/prices/PricesContractAdapter.ts:8]: Let's think about removing that layer as it doesn't have any custom logic

File: packages/sdk/src/contracts/ContractConnector.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { ContractAdapter } from "./ContractAdapter";
+
+export interface ContractConnector<A extends ContractAdapter> {
+  getAdapter(): Promise<A>;
+  getBlockNumber(rpcUrl: string): Promise<number>;
```
Comment (hatskier) [inline packages/sdk/src/contracts/ContractConnector.ts:5]: Maybe we can remove the rpcUrl

File: packages/starknet-connector/src/StarknetContractConnector.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      ["ACCEPTED_ON_L2", "REJECTED"]
+    );
+  }
+
+  getContract(): Contract {
+    return new Contract(this.getAbi(), this.contractAddress, this.provider);
+  }
+
+  async getBlockNumber(rpcUrl: string): Promise<number> {
+    const response = await axios.post(rpcUrl, {
```
Comment (hatskier) [inline packages/starknet-connector/src/StarknetContractConnector.ts:35]: Let's check if you can use this.provider to get block number

File: packages/sdk/src/contracts/prices/PricesCommandFactory.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { ContractDataProvider } from "../ContractDataProvider";
+import { ContractCommand } from "../ContractCommand";
+
+export interface PricesCommandFactory {
+  makeGetPricesCommand(
+    dataProvider: ContractDataProvider
+  ): ContractCommand<number[]>;
+
+  makeWritePricesCommand(
+    dataProvider: ContractDataProvider
```
[the remark this answers is not in the document]
Comment (Lukasz2891) [reply packages/sdk/src/contracts/prices/PricesCommandFactory.ts]: Enhancement :)

File: packages/sdk/src/contracts/ContractConnector.ts (hunk 2 above)
Comment (Lukasz2891) [reply packages/sdk/src/contracts/ContractConnector.ts:3]: The provider doesn't provide the rpcUrl :(

File: packages/starknet-connector/src/StarknetContractConnector.ts (hunk 3 above)
Comment (Lukasz2891) [reply packages/starknet-connector/src/StarknetContractConnector.ts:45]: It's possible to check data of the block passed as the argument, but there is not possible to check the last block data.

File: packages/sdk/src/contracts/ContractParamsProvider.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  async getPayloadData(): Promise<number[]> {
+    this.getDataFeedIds();
+
+    const payloadHex = await this.requestPayload(this.requestParams);
+
+    return Array.from(arrayify(payloadHex));
+  }
+
+  getFeedIds(): BigNumberish[] {
```
Comment (hatskier) [inline packages/sdk/src/contracts/ContractParamsProvider.ts:26]: I would change the name to smth like `getHexlifiedFeedIds` or `getFeedIdsBytes`

**Review by hatskier — APPROVED** [review]

**Pull request merged**